runtime/lua/vim/highlight.lua

```diff
@@ -9,28 +9,6 @@ M.priorities = {
   user = 200,
 }
 
----@private
-function M.create(higroup, hi_info, default)
-  local options = {}
-  -- TODO: Add validation
-  for k, v in pairs(hi_info) do
-    table.insert(options, string.format('%s=%s', k, v))
-  end
-  vim.cmd(
-    string.format(
-      [[highlight %s %s %s]],
-      default and 'default' or '',
-      higroup,
-      table.concat(options, ' ')
-    )
-  )
-end
-
----@private
-function M.link(higroup, link_to, force)
-  vim.cmd(string.format([[highlight%s link %s %s]], force and '!' or ' default', higroup, link_to))
-end
-
```

Comment:
Do not simply delete exported functions. Leave them as-is for now and add calls to `vim.deprecate`. For the "alternative" field, instruct users to use `vim.api.nvim_set_hl`:

```lua
vim.deprecate('vim.highlight.link', 'vim.api.nvim_set_hl', '0.9')
```